test/integration/functional_test.go

```diff
@@ -1520,6 +1523,42 @@ func validateServiceCmd(ctx context.Context, t *testing.T, profile string) {
 	}
 }
 
+func validateServiceCmdJSON(ctx context.Context, t *testing.T, profile string) {
+	// docs: Run `minikube service list -o JSON` and make sure the services are correctly listed as JSON output
+	t.Run("service_json_output", func(t *testing.T) {
+		targetSvcName := "hello-node"
+		// helper function to run command then, return target service object from json output.
+		extractServiceObjFunc := func(rr *RunResult) *service.SvcURL {
+			var jsonObjects service.URLs
+			err := json.Unmarshal(rr.Stdout.Bytes(), &jsonObjects)
+			if err != nil {
+				t.Errorf("failed to decode json from profile list: args %q: %v", rr.Command(), err)
+				return nil
+			}
+
+			for _, svc := range jsonObjects {
+				if svc.Name == targetSvcName {
+					return &svc
+				}
+			}
+			return nil
+		}
+
+		start := time.Now()
+		rr, err := Run(t, exec.CommandContext(ctx, Target(), "-p", profile, "service", "list", "-o", "json"))
+		elapsed := time.Since(start)
+		if err != nil {
+			t.Errorf("failed to list services with json format. args %q: %v", rr.Command(), err)
```

Comment:
This should be a `Fatalf` as well as we can't continue on if this command fails.

Reply:
Fixed!